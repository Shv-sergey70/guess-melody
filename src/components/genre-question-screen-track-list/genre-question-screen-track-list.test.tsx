import * as React from 'react';
import {configure, shallow} from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import GenreQuestionScreenTrackList from "./genre-question-screen-track-list";

configure({adapter: new Adapter()});

const answers = [
  {
    src: `test/track1.mp3`,
    genre: `pop`
  },
  {
    src: `test/track2.mp3`,
    genre: `jazz`
  },
  {
    src: `test/track3.mp3`,
    genre: `rock`
  },
  {
    src: `test/track4.mp3`,
    genre: `classic`
  }
];

test(`GenreQuestionScreenTrackList correctly renders`, () => {
  const renderFunction = (src, id) => <div>{src} - {id}</div>;
  const renderAudioPlayerMock = jest.fn()
    .mockImplementationOnce(renderFunction)
    .mockImplementationOnce(renderFunction)
    .mockImplementationOnce(renderFunction)
    .mockImplementationOnce(renderFunction);

  const tree = shallow(
      <GenreQuestionScreenTrackList
        activeItems={[false, true, false, true]}
        questionAnswers={answers}
        screenIndex={3}
        renderAudioPlayer={renderAudioPlayerMock}
        onChange={jest.fn()}
      />
  );

  expect(tree).toMatchSnapshot();
});
