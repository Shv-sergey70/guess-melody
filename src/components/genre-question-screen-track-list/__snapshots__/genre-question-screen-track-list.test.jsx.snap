// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreQuestionScreenTrackList correctly renders 1`] = `
Array [
  <GenreQuestionScreenTrack
    isSelected={false}
    key="3 - test/track1.mp3"
    onChange={[MockFunction]}
    src="test/track1.mp3"
    trackIndex={0}
  >
    <div>
      test/track1.mp3
       - 
      0
    </div>
  </GenreQuestionScreenTrack>,
  <GenreQuestionScreenTrack
    isSelected={true}
    key="3 - test/track2.mp3"
    onChange={[MockFunction]}
    src="test/track2.mp3"
    trackIndex={1}
  >
    <div>
      test/track2.mp3
       - 
      1
    </div>
  </GenreQuestionScreenTrack>,
  <GenreQuestionScreenTrack
    isSelected={false}
    key="3 - test/track3.mp3"
    onChange={[MockFunction]}
    src="test/track3.mp3"
    trackIndex={2}
  >
    <div>
      test/track3.mp3
       - 
      2
    </div>
  </GenreQuestionScreenTrack>,
  <GenreQuestionScreenTrack
    isSelected={true}
    key="3 - test/track4.mp3"
    onChange={[MockFunction]}
    src="test/track4.mp3"
    trackIndex={3}
  >
    <div>
      test/track4.mp3
       - 
      3
    </div>
  </GenreQuestionScreenTrack>,
]
`;
